buildpack.md

```diff
@@ -1102,17 +1101,27 @@ The `[[buildpack.licenses]]` table is optional and MAY contain a list of buildpa
 *Key: `sbom-formats = [ "<string>" ]`*
  - MUST be supported SBOM media types as described in [Software-Bill-of-Materials](#software-bill-of-materials).
 
-#### Stacks
+#### Targets
 
-A buildpack descriptor may specify `stacks`.
+A buildpack descriptor SHOULD specify `targets`.
 
-Each stack in `stacks` either:
-- MUST identify a compatible stack:
-   - `id` MUST be set to a [valid stack ID](https://github.com/buildpacks/spec/blob/main/platform.md#stack-id).
-   - `mixins` MAY contain one or more mixin names.
-- Or MUST indicate compatibility with any stack:
-   - `id` MUST be set to the special value `"*"`.
-   - `mixins` MUST be empty.
+Each target in `targets`:
+- MUST identify a compatible runtime environment:
+   - `os` and `arch` are required and MUST be valid identifiers as defined in the [OCI Image Specification](https://github.com/opencontainers/image-spec/blob/main/config.md)
+   - `variant` is optional and MUST be a valid identifier as defined in the [OCI Image Specification](https://github.com/opencontainers/image-spec/blob/main/config.md)
+   - `distributions` are optional and MUST describe the OS distributions supported by the buildpack
+     - For Linux-based images, `distributions.name` and `distributions.versions` SHOULD contain the values specified in `/etc/os-release` (`$ID` and `$VERSION_ID`), as the `os.version` field in an image config may contain combined distribution and version information
+     - For Windows-based images, `distributions.name` SHOULD be empty; `distributions.versions` SHOULD contain the value of `os.version` in the image config (e.g., `10.0.14393.1066`)
+
+If the `targets` list is empty, tools reading `buildpack.toml` will assume:
+  - `os = "linux"` and `arch = <matches any>` if `./bin/build` is present
+  - `os = "windows"` and `arch = <matches any>` if `./bin/build.bat` or `./bin/build.exe` are present
+  - `os = <matches any>` and `arch = <matches any>` for extensions
+
+Metadata specified in `[[targets]]` is validated against the runtime and build-time base images.
+* A buildpack target satisfies a base image target when:
+  * `os` and `arch` match, and
+  * Optional fields either match exactly, or are left blank by at least one target (blank values match any other value)
```

Comment:
```suggestion
  * Optional fields either match exactly, OR are left blank by at least one target (blank values match any other value), OR at least one optional field has the `*` value which matches any value in the corresponding field.
```

Reply:
Looking at `TargetSatisfiedForBuild` in the [lifecycle](https://github.com/buildpacks/lifecycle/blob/1ca5500ad911c1b636895f930aeca0a57559f21d/platform/target_data.go#L92) I don't think we handle `*` for distributions... should we update the implementation? I think for distributions there is always the option not to provide anything if we want to "match any" so `*` is not needed - but it should probably work the same way across fields, so that it's easier to understand.

Reply:
https://github.com/buildpacks/spec/pull/365#discussion_r1277588334